core/client/src/main/java/alluxio/client/file/policy/BlockLocationPolicy.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.policy;
+
+import alluxio.annotation.PublicApi;
+import alluxio.client.file.policy.options.BlockLocationPolicyCreateOptions;
+import alluxio.client.file.policy.options.BlockLocationPolicyGetWorkerOptions;
+import alluxio.util.CommonUtils;
+import alluxio.wire.WorkerNetAddress;
+
+/**
+ * <p>
+ * Interface for determining the Alluxio worker location to serve a block write or UFS block read.
+ * </p>
+ *
+ * <p>
+ * {@link alluxio.client.file.FileInStream} uses this to determine where to read a UFS block.
+ * </p>
+ *
+ * <p>
+ * A policy must have an empty constructor to be used as default policy.
+ * </p>
+ */
+@PublicApi
+public interface BlockLocationPolicy {
+
+  /**
+   * The factory for the {@link BlockLocationPolicy}.
+   */
+  class Factory {
+    private Factory() {} // prevent instantiation
+
+    /**
+     * Factory for creating {@link BlockLocationPolicy}.
+     *
+     * @param options the block location policy creation options
+     * @return a new instance of {@link BlockLocationPolicy}
+     */
+    public static BlockLocationPolicy create(BlockLocationPolicyCreateOptions options) {
```

Comment:
of the options class is moved to `alluxio.client.block.policy.options` then you could just call its class `CreateOptions`